### ISSUE TITLE
Get list of auto follow pattern throws a 404 not found

### ISSUE THREAD

**Issue by sebelga** [elastic/elasticsearch#35371]
**Elasticsearch version**: Current master branch

**Description of the problem including expected versus actual behavior**:

When we query to retrieve the autofollow patterns (`GET /_ccr/auto_follow`) on a cluster that **never had any previous auto follow pattern**, a 404 error is thrown

```
{
    "error": {
        "root_cause": [
            {
                "type": "resource_not_found_exception",
                "reason": "auto-follow pattern [null] is missing"
            }
        ],
        "type": "resource_not_found_exception",
        "reason": "auto-follow pattern [null] is missing"
    },
    "status": 404
}

```

If we then CREATE and then DELETE an auto follow pattern, and then make the above query, we receive the correct `200` status with an empty object response `{}`

**Steps to reproduce**:

Please include a *minimal* but *complete* recreation of the problem, including
(e.g.) index creation, mappings, settings, query etc.  The easier you make for
us to reproduce it, the more likely that somebody will take the time to look at it.

 1. Create a new Elasticsearch Cluster and launch it
 2. Activate trial license: `POST /_xpack/license/start_trial?acknowledge=true`
 3. Query auto_follow patterns:  `GET /_ccr/auto_follow`

**Comment by martijnvg** [comment]
Thanks for raising this bug @sebelga! This now fixed via #35381

**Comment by sebelga** [comment]
Great, thanks @martijnvg !
